src/main/java/cardibuddy/model/CardiBuddy.java

```diff
@@ -52,31 +64,68 @@ public void setFlashcards(List<Flashcard> flashcards) {
     public void resetData(ReadOnlyCardiBuddy newData) {
         requireNonNull(newData);
 
+        setDecks(newData.getDeckList());
         setFlashcards(newData.getFlashcardList());
     }
 
-    //// person-level operations
+    //// deck-level operations
+    /**
+     * Returns true if a deck with the same identity as {@code deck} exists in cardibuddy.
+     */
+    public boolean hasDeck(Deck deck) {
+        requireNonNull(deck);
+        return decks.contains(deck);
+    }
+
+    /**
+     * Adds a deck to the cardi buddy.
+     * The deck must not already exist in the cardi buddy.
+     */
+    public void addDeck(Deck d) {
+        decks.add(d);
+    }
+
+    /**
+     * Replaces the given deck {@code target} in the list with {@code editedDeck}.
+     * {@code target} must exist in the cardi buddy.
+     * The deck identity of {@code editedFlashcard} must not be the same as another existing deck in the cardi buddy.
+     */
+    public void setDeck(Deck target, Deck editedDeck) {
+        requireNonNull(editedDeck);
+
+        decks.setDeck(target, editedDeck);
+    }
 
     /**
-     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     * Removes {@code key} from this {@code CardiBuddy}.
+     * {@code key} must exist in the cardi buddy.
      */
-    public boolean hasFlashcard(Flashcard person) {
-        requireNonNull(person);
-        return flashcards.contains(person);
+    public void removeDeck(Deck key) {
```

Comment:
Sorry for the confusion but I was wondering why the variable name is named key? Is it not just a normal deck being removed?